usr/src/cmd/mdb/i86pc/modules/unix/sec.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * CDDL HEADER
+ *
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source. A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2018 OmniOS Community Edition (OmniOSce) Association.
+ */
+
+#include <limits.h>
+#include <sys/mdb_modapi.h>
+#include <sys/sysinfo.h>
+#include <sys/sunmdi.h>
+#include <sys/x86_archext.h>
+
+int
+/* LINTED E_FUNC_ARG_UNUSED */
+sec_dcmd(uintptr_t addr, uint_t flags, int argc, const mdb_arg_t *argv)
+{
+	uint64_t kpti_enable;
+	int x86_use_pcid, x86_use_invpcid;
+	void *fset;
+	GElf_Sym sym;
+	size_t sz;
+	int opt_p = FALSE;
+
+	if (mdb_getopts(argc, argv, 'p', MDB_OPT_SETBITS, TRUE, &opt_p) != argc)
+		return (DCMD_USAGE);
+
+	/* Meltdown (CVE-2017-5754) */
+
+	if (mdb_readvar(&kpti_enable, "kpti_enable") == -1)
+		kpti_enable = 2;
+	if (mdb_readvar(&x86_use_pcid, "x86_use_pcid") == -1)
+		x86_use_pcid = -1;
+	if (mdb_readvar(&x86_use_invpcid, "x86_use_invpcid") == -1)
+		x86_use_invpcid = -1;
+
+	sz = sizeof (uchar_t) * BT_SIZEOFMAP(NUM_X86_FEATURES);
+	fset = mdb_zalloc(sz, UM_NOSLEEP);
```

Comment:
We should handle NULL case as UM_NOSLEEP is being used.
UM_NOSLEEP
If sufficient memory to fulfill the request is not immediately available, return NULL to indicate failure. The caller must check for NULL and handle this case appropriately.